coderd/coderd.go

```diff
@@ -1427,11 +1424,14 @@ func (api *API) Close() error {
 		api.updateChecker.Close()
 	}
 	_ = api.workspaceAppServer.Close()
+	_ = api.agentProvider.Close()
```

Comment:
can you add a comment here about why the close happens in this order?

Reply:
This was Dean's change so I honestly don't know why

Reply:
It was just to close the agent provider before the coordinator, the other way around makes less sense.

Reply:
We should be documenting why things get shut down in particular orders, so that future editors of this function know what they can reorder and what they cannot.